### PR TITLE
Move grpc stub to background thread

### PR DESCRIPTION
**Pull request by ksheremet**

### CODE REVIEW

File: app/src/main/java/ch/sheremet/dasfoo/rover/android/client/MainActivity.java [hunk 1]
```diff
@@ -37,6 +38,24 @@
     private EditText mPortEdit;
     private TextView mResultText;
 
+    public final int getPort() {
+        String port = mPortEdit.getText().toString();
+        if (TextUtils.isEmpty(port)) {
+            Toast.makeText(this, "You did not enter a port", Toast.LENGTH_SHORT).show();
```
Comment (dotdoom) [inline app/src/main/java/ch/sheremet/dasfoo/rover/android/client/MainActivity.java]: How about throw an exception here, and have the caller show the toast in `catch { ... }`?
Then `notEmptyHostAndPort` becomes unnecessary.

Comment (ksheremet) [inline app/src/main/java/ch/sheremet/dasfoo/rover/android/client/MainActivity.java]: You're right. I need to throw a throwable object. I chose MissingFormatArgumentException. It looks appropriate for this.

**Comment by dotdoom** [comment]
LGTM

File: app/src/main/java/ch/sheremet/dasfoo/rover/android/client/MainActivity.java [hunk 2]
```diff
@@ -132,6 +134,15 @@ protected void onPreExecute() {
         @Override
         protected String doInBackground(final AbstractGrpcTaskExecutor... params) {
             checkProviderInstaller();
+            try {
+                if (mGrpcConnection == null
+                        || getHost().equals(mGrpcConnection.getHost())
```
Comment (dotdoom) [inline app/src/main/java/ch/sheremet/dasfoo/rover/android/client/MainActivity.java]: Should this be `!getHost().equals...` ?

**Comment by dotdoom** [comment]
LGTM

[Approved with PullApprove](https://pullapprove.com/dasfoo/rover-android-client/pull-request/27/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=dasfoo/rover-android-client)

**Comment by ksheremet** [comment]
LGTM

[Approved with PullApprove](https://pullapprove.com/dasfoo/rover-android-client/pull-request/27/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=dasfoo/rover-android-client)

**Pull request merged**